pkg/lib/registry/registry.go

```diff
@@ -342,9 +342,45 @@ func (r RegistryUpdater) DeprecateFromRegistry(request DeprecateFromRegistryRequ
 		return fmt.Errorf("unable to migrate database: %s", err)
 	}
 
-	deprecator := sqlite.NewSQLDeprecatorForBundles(dbLoader, request.Bundles)
+	// Check if all bundlepaths are valid
+	var toDeprecate []string
+
+	dbQuerier := sqlite.NewSQLLiteQuerierFromDb(db)
+	registryBundles, err := dbQuerier.ListBundles(context.TODO())
+	if err != nil {
+		if !request.Permissive {
+			return fmt.Errorf("cannot verify bundle presence in registry: %v", err)
+		}
+		toDeprecate = request.Bundles
+	}
+
+	if registryBundles != nil {
```

Comment:
Is this the right conditional check? Should we be checking the length of this array instead?

Reply:
I almost wonder whether this logic would be served being contained in a helper function that we can instead call in this method. It would be easier to add unit tests to hit this case so we don't regress in future releases.

Reply:
It's a edge case we'd almost never run into, but if you try to deprecate a bundle from an empty registry, it wouldn't complain about missing bundles if we look for length.